build/patches/Bromite-AdBlockUpdaterService.patch

```diff
@@ -1224,19 +1557,27 @@ new file mode 100644
 +    return;
 +  }
 +
-+  // ignoring 'headers' as 'Last-Modified' has already been picked up by OnResponseStarted
-+  const base::TimeDelta dt =
-+          last_modified_ - min_last_modified_;
++  int net_error = simple_url_loader_->NetError();
++  simple_url_loader_.reset();
```

Comment:
ah, one more thing: do you ever know that I am in doubt if quic is really disabled? When digging, I've seen here and there some #if **no android** when ask if disabled.

I don't know exactly the protocol (I will study it) but from what I understood the problem was in the session cache, which would allow the host to track you if it is not deleted, because it would keep the local cryptographic key, obviously different for client and therefore with possibility tracking. I am wrong? You know something about this?

between the flags there is a way to disable it, and I thought it was applied, but with those #if they make me doubt.

Reply:
> I have repeatedly tried in different ways to force the socket to close but I have not succeeded, it remains open until the android recents open. I think it's standard chromium behavior, but I don't like it, I think I open an issue to remind me to check.

It probably goes back in a connection pool and is closed automatically when not needed anymore; why do you think it is a problem? For privacy reasons?



> I am in doubt if quic is really disabled

It is not disabled.

> I don't know exactly the protocol (I will study it) but from what I understood the problem was in the session cache, which would allow the host to track you if it is not deleted, because it would keep the local cryptographic key, obviously different for client and therefore with possibility tracking. I am wrong? You know something about this?

It was disabled for a while in Bromite, then Chromium fixed/mitigated the problem you are mentioning and the patch was dropped; see https://github.com/bromite/bromite/blob/master/CHANGELOG.md#830410346

When I dropped the patch I checked about the remaining concerns and there were none based on facts; also, QUIC is going to be replaced by HTTP/3 and Chromium implementation is aligned with that as far as I know.

If you find that there are still concerns we can look into this again, but there was no privacy concern left back then.

Reply:
> It probably goes back in a connection pool and is closed automatically

certainly, but it seems too aggressive

> For privacy 

from what I have seen, the socket stays open for my entire browsing session and there is nothing that makes it close except by opening the recents, so theoretically github (via adblock updater) knows when I use bromite.
This particular situation is resolved with my changes, because, at the very least, now github is notified every 7 days (as much as entered in `next_check_delay`), and if the user really bothers this too, he can set up his own filter mirror

however it is not a problem of this patch, even the favicons sockets (opened by the new tab page, which fortunately I don't use) remain open.

The anomaly is that by clearing the cache (therefore, in the belief to erase all traces), chromium requests immediately the favicons for all sites in newtabs, reopening the sockets and never closing them: so, theoretically, all the sites that I would have in the newtab know when I delete the history! yes, they don't send any cookies (and I hope not even the etag), but still I'm sending some sort of ping, and would be even worse if it were true the quic issue.

All this, however, it's my thought, should be better documented and verified, but it should make us think why in anonymous browsing favicons are never loaded

> It is not disabled.

i disabled it via chrome://flags

> If you find that there are still concerns we can look into this again

i think i will check

Reply:
> the socket stays open for my entire browsing session and there is nothing that makes it close except by opening the recents

What are the "recents"?

> github is notified every 7 days

GitHub is not notified, the collection is passive; HTTP requests are logged and they surely will analyse traffic in bulk to see usage patterns, cost analysis and to provide data to agencies during investigations, as needed.

See https://docs.github.com/en/github/site-policy/github-privacy-statement#usage-information; I could not find information about for how long they keep logs though.

> reopening the sockets and never closing them

This cannot be; if it is true, you can report it as a bug to Chromium. I am afraid you are misinterpreting the TCP connection states. What is it exactly that you see? Do you see any TCP/UDP packets going back and forth during this time when sockets are never closed?

> it should make us think why in anonymous browsing favicons are never loaded

Source?

> I hope not even the etag

With the recent enabling in Bromite of the network isolation features (see https://github.com/bromite/bromite/issues/836) the concerns about shared connections are mitigated.

Reply:
> What are the "recents"?

I mean https://developer.android.com/guide/components/activities/recents

> GitHub is not notified, the collection is passive

yes, sure, sorry, "notified" is wrong

> Do you see any TCP/UDP packets going back and forth during this time when sockets are never closed?

no, only tcp port open, no traffic

> What is it exactly that you see?

maybe I do a report, I can always be wrong

> Source?

ok, sorry, since the ui doesn't show the icons, i guess they don't load. I haven't checked the code yet.

> With the recent enabling in Bromite of the network isolation features (see #836) the concerns about shared connections are mitigated. As for ETag, you can easily verify this.

perfect, I'll check

Reply:
> no, only tcp port open, no traffic
> maybe I do a report, I can always be wrong

Are keep-alive packets being sent? It might keep the connections open to avoid re-doing the TLS handshake.

Reply:
> Are keep-alive packets being sent? It might keep the connections open to avoid re-doing the TLS handshake.

is the first thing I thought, but `simple_url_loader` has no option to control that. 
I have the trace of the net, I have to check. but above all I have to check because opening the recents the sockets are closed, maybe somewhere i find a timeout to configure.

Reply:
If there are no TCP packets transferred to or received from the host, then it is not an issue; you can verify this with Wireshark.

Reply:
> then it is not an issue

it depends, in my opinion maintaining an active tcp session is an issue, with or without traffic, especially if it remains open until the browser is closed.
however, let me check with chromium net-internals as soon as I have some time.

Reply:
Again: without traffic, there is no issue. You might be confused by kernel optimisations that leave the port "pre-opened" for the next connection.